### PR TITLE
Fixes human burgers not naming after the donor

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

Instead of human burgers always being called human burgers they'll try to name themselves after the donor or their job title again.

In the process, does the following:
- Adds name and job of meat source to mob_meat material 
- Transfers custom materials to the results of processed atoms
- Transfers custom materials to the results of grilling

Also makes processed atoms have the same pixel offsets as their source while randomising any subsequent ones.

## Why It's Good For The Game

Human burgers are a staple of SS13 cuisine and this returns them to their former glory.

If you use a rolling pin on a meatball the resulting patty won't have moved randomly.

## Changelog
:cl: cacogen
fix: Human burgers can name themselves after their ingredient sources again
/:cl:

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1435,8 +1435,13 @@
 		var/list/atom/created_atoms = list()
 		for(var/i = 1 to chosen_option[TOOL_PROCESSING_AMOUNT])
 			var/atom/created_atom = new atom_to_create(drop_location())
-			created_atom.pixel_x = rand(-8, 8)
-			created_atom.pixel_y = rand(-8, 8)
+			if(custom_materials)
+				created_atom.set_custom_materials(custom_materials, 1 / chosen_option[TOOL_PROCESSING_AMOUNT])
```
Comment (Qustinnus) [inline code/game/atoms.dm:1439]: this'll have a bunch of unrelated changes, why do you need this

Comment (cacogen) [reply code/game/atoms.dm:1439]: It seems like a less snowflakey solution than adding ownership vars to all the products between meat and burger and then adding special code in situations like this to pass them on. 

Adding vars storing the source's name and job to the mob_meat material seemed like the exact sort of thing you intended the material system to be used for. I was originally going to have everything made of mob_meat name itself after the owner (e.g. "Leroy Coldsmith meat toilet") but there were issues with some of the names.

It seems like an oversight (or unfinished) that materials are passed through the food processor but not through tool processing or grilling. At the moment custom materials from raw meat passes into raw meatballs from the food processor but disappears after the meatball is flattened into a patty or grilled. 

What consequences do you think this will have, anyway? Unless you have something specific in mind I would just wait and see what happens and fix what arises, if anything.

**Comment by Pugie** [comment]
donor implies consent

**Comment by maxymax13** [comment]
> 
> 
> donor implies consent

if they didnt want to be a burger maybe they should have simply not let me gib them
